domain/src/main/java/org/oppia/android/domain/oppialogger/analytics/PerformanceMetricsController.kt

```diff
@@ -0,0 +1,178 @@
+            .build()
+        } else {
+          // TODO(#1433): Refactoring for logging exceptions to both console and exception loggers.
+          val exception =
+            IllegalStateException(
+              "Least Recent Event index absent -- MetricLogStorageCacheSize is 0"
+            )
+          consoleLogger.e(
+            "PerformanceMetricsController",
+            "Failure while caching metric log.",
+            exception
+          )
+          exceptionLogger.logException(exception)
+        }
+      }
+      return@storeDataAsync oppiaMetricLogs.toBuilder().addOppiaMetricLog(oppiaMetricLog).build()
+    }.invokeOnCompletion {
+      it?.let { consoleLogger.e("PerformanceMetricsController", "Failed to store metric log.", it) }
+    }
+  }
+
+  /** Returns a metric log containing relevant data for metric log reporting. */
+  private fun createMetricLog(
+    timestamp: Long,
+    priority: Priority,
+    currentScreen: OppiaMetricLog.CurrentScreen,
+    loggableMetric: OppiaMetricLog.LoggableMetric
+  ): OppiaMetricLog {
+    return OppiaMetricLog.newBuilder().apply {
+      this.timestampMillis = timestamp
+      this.priority = priority
+      this.currentScreen = currentScreen
+      this.loggableMetric = loggableMetric
+      this.isAppInForeground = performanceMetricsUtils.isAppInForeground()
+      this.storageTier = performanceMetricsUtils.getDeviceStorageTier()
+      this.memoryTier = performanceMetricsUtils.getDeviceMemoryTier()
+    }.build()
+  }
+
+  /**
+   * Returns the index of the least recent event from the existing store on the basis of recency and
+   * priority.
+   *
+   * At first, it checks the index of the least recent event which has LOW priority. If that
+   * returns null, then it checks the index of the least recent event which has MEDIUM priority. If
+   * that returns null, then it checks the index of the least recent event regardless of the
+   * priority is returned.
+   */
+  private fun getLeastRecentMetricLogIndex(oppiaMetricLogs: OppiaMetricLogs): Int? =
+    oppiaMetricLogs.oppiaMetricLogList.withIndex()
+      .filter { it.value.priority == Priority.LOW_PRIORITY }
+      .minByOrNull { it.value.timestampMillis }?.index ?: getLeastRecentMediumPriorityEventIndex(
+      oppiaMetricLogs
+    )
+
+  /**
+   * Returns the index of the least recent event from the existing store on the basis of recency and
+   * priority.
+   *
+   * At first, it checks the index of the least recent event which has LOW priority. If that
+   * returns null, then it checks the index of the least recent event regardless of the
+   * priority is returned.
+   */
+  private fun getLeastRecentMediumPriorityEventIndex(oppiaMetricLogs: OppiaMetricLogs): Int? =
+    oppiaMetricLogs.oppiaMetricLogList.withIndex()
+      .filter { it.value.priority == Priority.MEDIUM_PRIORITY }
+      .minByOrNull { it.value.timestampMillis }?.index ?: getLeastRecentGeneralEventIndex(
+      oppiaMetricLogs
+    )
```

Comment:
```suggestion
      .minByOrNull { it.value.timestampMillis }?.index
      ?: getLeastRecentGeneralEventIndex(oppiaMetricLogs)
```

We should wrap at a higher syntactical element.Ditto elsewhere for similar structure.

Reply:
got it, done!